chipsec/utilcmd/tpm_cmd.py

```diff
@@ -11,27 +12,77 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU General Public License for more details.
 #
+#You should have received a copy of the GNU General Public License
+#along with this program; if not, write to the Free Software
+#Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+#Contact information:
+#chipsec@intel.com
+#
 
 from chipsec.command import BaseCommand
 from chipsec.hal import tpm_eventlog
+from chipsec.hal import tpm
 
 class TPMCommand(BaseCommand):
-    """
-    >>> chipsec_util tpm parse_log <file>
+        """
+        >>> chipsec_util tpm parse_log <file>
+        >>> chipsec_util tpm state <locality>
+        >>> chipsec_util tpm command <commandName> <locality> <command_parameters>
 
-    Examples:
+        locality: 0 | 1 | 2 | 3 | 4
+        commands - parameters:
+        pccrread - pcr number ( 0 - 23 )
+        nvread - Index, Offset, Size
+        startup - startup type ( 1 - 3 )
+        continueselftest
+        getcap - Capabilities Area, Size of Sub-capabilities, Sub-capabilities
+        forceclear
+        
+        Examples:
+                        
+        >>> chipsec_util tpm parse_log binary_bios_measurements
+        >>> chipsec_util tpm state 0
+        >>> chipsec_util tpm command pcrread 0 17
+        >>> chipsec_util tpm command continueselftest 0
+        """
+        no_driver_cmd = ['parse_log']
 
-    >>> chipsec_util tpm parse_log binary_bios_measurements
-    """
-    def requires_driver(self):
-        return False
+        def requires_driver(self):
+            if len(self.argv) < 4:
+                return False
+            if self.argv[2] in self.no_driver_cmd:
+                return False
+            return True
 
-    def run(self):
-        if len(self.argv) < 4:
-            print TPMCommand.__doc__
-            return
-        log = open(self.argv[3])
-        tpm_eventlog.parse(log)
+        def run(self):
+            try:
+                _tpm = tpm.TPM(self.cs)
+            except TpmRuntimeError, msg:
+                print(msg)
+                return
 
```

Comment:
It looks like we have some extra indentation here...  Same with the chunk above.  I will pull the code to check.